cmd/infra/aws/route53.go

```diff
@@ -146,11 +128,65 @@ func (o *DestroyInfraOptions) CleanupPublicZone(ctx context.Context, client rout
 		if !isRoute53RecordNotFoundErr(err) {
 			return fmt.Errorf("failed to delete wildcard record from public zone %s: %w", id, err)
 		}
-		log.Info("Wildcard record not found in public zone")
-
 	} else {
-		log.Info("Deleted wildcard record from public zone", "id", id, "name", recordName)
+		log.Info("Deleted wildcard record from public hosted zone", "id", id, "name", recordName)
+	}
+	return nil
+}
+
+func deleteZone(ctx context.Context, id string, client route53iface.Route53API) error {
+	err := deleteRecords(ctx, client, id)
+	if err != nil {
+		return fmt.Errorf("failed to delete hosted zone records: %v", err)
+	}
+	if _, err = client.DeleteHostedZoneWithContext(ctx, &route53.DeleteHostedZoneInput{
+		Id: aws.String(id),
+	}); err != nil {
+		return fmt.Errorf("failed to delete hosted zone: %v", err)
+	}
+	return nil
+}
+
+func deleteRecords(ctx context.Context, client route53iface.Route53API, id string) error {
+	lrrsi := &route53.ListResourceRecordSetsInput{
+		HostedZoneId: aws.String(id),
+	}
+	output, err := client.ListResourceRecordSetsWithContext(ctx, lrrsi)
+	if err != nil {
+		return err
+	}
+	if len(output.ResourceRecordSets) == 0 {
+		return nil
+	}
+	var changeBatch route53.ChangeBatch
+	var deleteRequired bool
+	for _, rrs := range output.ResourceRecordSets {
+		if *rrs.Type == "NS" || *rrs.Type == "SOA" {
+			continue
+		}
```

Comment:
This seems to assume that after `deleteRecords()` only the _default_ NS and SOA will remain, allowing `DeleteHostedZone` to proceed:

>You can delete a hosted zone only if it contains only the default SOA record and NS resource record sets. If the hosted zone contains other resource record sets, you must delete them before you can delete the hosted zone. 

1. Are there any circumstances where non-default NS/SOA records can end up in here?
2. If we're trying to tear down the private zone, is there harm in deleting the NS/SOA records (e.g. a zone shared by multiple clusters?)